### PR TITLE
Updated Phenotype modal

### LINKED ISSUE #294: Update phenotype modal with new resource data
@jeffmax What are the new fields/data that need to be displayed?

### PR DESCRIPTION
**Pull request by hassanNS**
Fix #294

The new look (top part is still the same):
screen shot 2014-05-15 at 1 24 46 pm

Signed-off-by: Sheik Hassan solergiga@yahoo.com

### CODE REVIEW

**Comment by mitalia** [comment]
That thumbnail is awful tiny. Is that the default size?

**Comment by hassanNS** [comment]
@mitalia It's a decent size (around 200x200 i think) I wasn't sure where to put it.

**Comment by mitalia** [comment]
Instead of burying it in the footer, it could go right at the top at full size with the link next to it, followed by all the other info.

**Comment by naegelyd** [comment]
I don't think things in this display should have ellipses. They should be allowed to take up the space they need. The modal can be made to scroll if more space is needed. Also the modal itself could be made bigger. Since this is a screenshot, I am not sure if the data under "Procedures" actually contains `There are ...` or if the ellipsis is there because of overflow. If `There are ...` is the actual content then feel free to ignore this comment.

**Comment by hassanNS** [comment]
New look with the updated thumbnail. If it gets any bigger it gets very blurry
screen shot 2014-05-15 at 6 00 06 pm

**Comment by naegelyd** [comment]
Are the thumbnails known to be a fixed size when they are retrieved from
the pedigree endpoint? If so, you should add CSS rules to fix the displayed
thumbnail image to those exact dimensions. This should eliminate any
blurriness.

**Comment by jeffmax** [comment]
So I restrict the width to 150 pixels. The dimensions are highly variable depending on the size of the pedigree, so I let it keep the ratio for the height.

**Comment by mitalia** [comment]
If possible, I would suggest moving the link above the pedigree and changing text to say, "View full-size pedigree" or "View pedigree PDF"

**Comment by hassanNS** [comment]
screen shot 2014-05-16 at 9 46 00 am

The width has been limited to the default dimensions of the pedigree

File: varify/static/js/src/ui/modals/phenotype.js [hunk 1]
```diff
@@ -168,6 +174,91 @@ define([
...
+            html.push('<div class=span7>');
+            html.push('<h6>Family hx:</h6>' + attr.family_hx); // jshint ignore: line
+            html.push('</div>');
+
+            html.push('<div class=span7>');
+            html.push('<h6>Medical hx:</h6>' + attr.medical_hx); // jshint ignore: line
+            html.push('</div>');
+
+            return html.join('');
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Remove blank line

File: varify/static/js/src/ui/modals/phenotype.js [hunk 2]
```diff
@@ -168,6 +174,91 @@ define([
             return html.join('');
         },
 
+        renderFindings: function(attr) {
+            var html = [];
+
+            html.push('<div class=span9>');
+            html.push('<h6>Imaging Studies:</h6>' + attr.imaging_studies); // jshint ignore: line
+            html.push('</div>');
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: This html construction is pretty trivial and can probably just be done within the template. This comment applies to this window as a whole. As an example, look at the new result details modal. [Here](https://github.com/cbmi/varify/blob/279478179116cab5ab27aaaf73d6d037b8150ded/varify/static/js/src/ui/variant/summary.js) is a view that uses a [template](https://github.com/cbmi/varify/blob/279478179116cab5ab27aaaf73d6d037b8150ded/varify/static/templates/variant/summary.html) to render the data rather than construction the html within the view itself like it is done here.

File: varify/static/stylesheets/scss/style.scss [hunk 3]
```diff
@@ -283,3 +291,5 @@ body, #concept-panel, #context-panel {
         border-top: 1px solid #eee;
     }
 }
+
```
Comment (naegelyd) [inline varify/static/stylesheets/scss/style.scss]: Remove these blank lines

File: varify/static/js/src/ui/modals/phenotype.js [hunk 4]
```diff
@@ -168,56 +182,51 @@ define([
...
+            // Initially, the model is not set to anything. It is only after the
+            // request for the phenotype is successful that we have a model.
+            if (this.model) {attr=this.model.attributes;} else return;
 
             this.ui.annotations.html(this.renderHPO(attr.hpoAnnotations));
             this.ui.notes.html(this.renderNotes(attr.notes));
             this.ui.diagnoses.html(this.renderDiagnoses(attr));
-            this.ui.updateTimes.html(this.renderUpdateTimes(attr));
 
             if (attr.pedigree) {
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: This if/else block can all be moved to the template. See [this](https://github.com/cbmi/varify/blob/279478179116cab5ab27aaaf73d6d037b8150ded/varify/static/templates/variant/effect-item.html#L7-L9) as an example of only showing links when an attribute exists.

Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js]: You're right. Didn't notice that

File: varify/static/js/src/ui/modals/phenotype.js [hunk 5]
```diff
@@ -8,20 +8,26 @@ define([
...
 
     var Phenotype = Marionette.ItemView.extend({
-        className: 'modal hide',
+        className: 'phenotype-detail-modal modal hide',
 
         template: 'varify/modals/phenotype',
 
+
         ui: {
             annotations: '[data-target=annotations]',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: With the template handling a lot of the data itself, I imagine a lot of these are unnecessary now. Remove any of these ui elements that are no longer needed

File: varify/static/js/src/ui/modals/phenotype.js [hunk 6]
```diff
@@ -168,56 +182,51 @@ define([
...
-            var attr = model.attributes;
+            var attr = {};
+
+            // Initially, the model is not set to anything. It is only after the
+            // request for the phenotype is successful that we have a model.
+            if (this.model) {attr=this.model.attributes;} else return;
 
             this.ui.annotations.html(this.renderHPO(attr.hpoAnnotations));
             this.ui.notes.html(this.renderNotes(attr.notes));
             this.ui.diagnoses.html(this.renderDiagnoses(attr));
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: If annotations, notes, and diagnoses are too complex for the template then they can be made into regions and this class can be made to extend from marionette.Layout instead of itemview. An example of a similar case are the lists(effects, phenotypes, etc) on the [results dialog](https://github.com/cbmi/varify/blob/variant-sets/varify/static/js/src/ui/modals/result.js). The goal of this refactor is to remove all the complex html construction from this class and to take full advantage of templating and marionette classes.

File: varify/static/js/src/ui/modals/phenotype.js [hunk 7]
```diff
@@ -8,20 +8,26 @@ define([
 ], function(_, Marionette, models, utils) {
 
     var Phenotype = Marionette.ItemView.extend({
-        className: 'modal hide',
+        className: 'phenotype-detail-modal modal hide',
 
         template: 'varify/modals/phenotype',
 
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Remove blank line

**Comment by naegelyd** [comment]
Move the new templates to a subfolder in `modals` called `phenotype`

File: varify/static/js/src/main.js [hunk 8]
```diff
@@ -58,6 +62,10 @@ require({
     c.templates.set('varify/tables/header', header);
     c.templates.set('varify/modals/result', result);
     c.templates.set('varify/modals/phenotype', phenotype);
+    c.templates.set('varify/modals/annotations', annotations);
```
Comment (naegelyd) [inline varify/static/js/src/main.js]: These should also be renamed to `varify/modals/phenotype/*`

File: varify/static/js/src/ui/modals/details.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+/* global define */
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details.js]: Rename this file to phenotype.js

File: varify/static/js/src/ui/modals/details.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+define([
+    'underscore',
+    './details/annotations',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details.js]: these should be in a subfolder called phenotype

File: varify/static/js/src/ui/modals/details/Annotations.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+/* global define */
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: lower case file name

File: varify/static/js/src/ui/modals/details/Annotations.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+define([
+    'underscore',
+    'marionette'
+], function(_, Marionette) {
+
+    var AnnotationItem = Marionette.ItemView.extend({
+        tagName: 'li',
+        template: 'varify/modals/annotationsItem',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: Add blank line above, remove trailing comma

File: varify/static/js/src/ui/modals/details/Annotations.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    });
+
+    var Annotations = Marionette.CompositeView.extend({
+        template: 'varify/modals/annotations',
+        itemView: AnnotationItem,
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: blank line abvoe

File: varify/static/js/src/ui/modals/details/Annotations.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        itemViewContainer: '[data-target=annotation-list]',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: blank line above

File: varify/static/js/src/ui/modals/details/Annotations.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+        serializeData: function() {
+            var collection = false;
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: What is the reasoning behind this method? Is it something to do with the template?

File: varify/static/js/src/ui/modals/details/diagnoses.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+], function(_, Marionette) {
+
+    var DiagnoseItem = Marionette.ItemView.extend({
+        tagName:'li',
+        template: 'varify/modals/diagnosesItem',
+    });
+
+    var Diagnoses = Marionette.CompositeView.extend({
+        template: 'varify/modals/diagnoses',
+        itemView: DiagnoseItem,
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/diagnoses.js]: blank line above

File: varify/static/js/src/ui/modals/details/diagnoses.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        itemViewContainer: '[data-target=items]',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/diagnoses.js]: blank line above

File: varify/static/js/src/ui/modals/phenotype.js [hunk 18]
```diff
@@ -2,32 +2,40 @@
...
-            updateTimes: '[data-target=update-times]',
             warning: '[data-target=warning]',
             hideOnRetrieve: '[data-action=hide-on-retrieve]'
         },
 
+        regions: {
+            confirmedDiagnoses: '[data-target=confirmed-diagnoses]',
+            suspectedDiagnoses: '[data-target=suspected-diagnoses]',
+            ruledOutDiagnoses: '[data-target=ruled-out-diagnoses]',
+            annotations: '[data-target=annotations]',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: remove trailing comma

File: varify/static/js/src/ui/modals/phenotype.js [hunk 19]
```diff
@@ -60,46 +68,32 @@ define([
...
+            _.bindAll(this, 'onFetchError');
 
             if (!(this.data.context = this.options.context)) {
                 throw new Error('context model required');
             }
         },
 
-        renderHPO: function(annotations) {
-            var html = [], key, priority;
+        serializeData: function() {
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: What value is this method adding over the [normal implementation](https://github.com/marionettejs/backbone.marionette/blob/master/src/marionette.itemview.js#L22-L31)?

File: varify/static/js/src/ui/modals/phenotype.js [hunk 20]
```diff
@@ -119,105 +113,74 @@ define([
...
-            return html.join('');
-        },
+            // Render each region
+            this.annotations.show( new details.Annotations({
+                collection: this.populateHPOcollection(attr.hpoAnnotations)
+            }));
 
-        onFetchSuccess: function(model, response, recalculateRankings) {
-            delete this.request;
+            this.confirmedDiagnoses.show( new details.Diagnoses({
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: remove space after (

File: varify/static/js/src/ui/modals/phenotype.js [hunk 21]
```diff
@@ -119,105 +113,74 @@ define([
...
+                name: 'Confirmed Diagnoses',
+                collection: this._renderDiagnoses(attr.confirmedDiagnoses)
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Why is the collection being set to the result of a render method?

File: varify/static/js/src/ui/modals/phenotype.js [hunk 22]
```diff
@@ -250,8 +213,13 @@ define([
...
                     processData: true,
-                    success: function(model, response) {
-                        _this.onFetchSuccess(model, response, recalculateRankings);
+                    success: function(model) {
+                        // Set the the model so that the serializeData function
+                        // has access to the attributes.
+                        _this.model = model;
+                        _this.render();
+                        _this.ui.headerLabel.text('Phenotypes for ' + samples[0]);
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: remove blank line

File: varify/static/js/src/ui/modals/phenotype.js [hunk 23]
```diff
@@ -264,7 +232,8 @@ define([
 
                 this.ui.error.show();
             }
-        },
+         },
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Revert this change, remove the blank line and the space

File: varify/static/templates/modals/annotations.html [hunk 24]
```diff
@@ -0,0 +1,5 @@
+<h6>HPO</h6>
+<ul data-target=annotation-list></ul>
+<% if (data.annotation == false) { %>
```
Comment (naegelyd) [inline varify/static/templates/modals/annotations.html]: Set [emptyView](https://github.com/marionettejs/backbone.marionette/blob/master/docs/marionette.collectionview.md#collectionviews-emptyview) on your compositeviews rather than doing this manual checking for empty collections.

File: varify/static/templates/modals/annotationsItem.html [hunk 25]
```diff
@@ -0,0 +1,4 @@
+<a href='<%= "http://www.human-phenotype-ontology.org/hpoweb/showterm?id=" + data.hpo_id %>' "target"="_blank" > <%= data.name %> </a>
```
Comment (naegelyd) [inline varify/static/templates/modals/annotationsItem.html]: Why is target quoted here?

Comment (naegelyd) [inline varify/static/templates/modals/annotationsItem.html]: Remove space before and after `>` after target attribute set

File: varify/static/templates/modals/diagnosesItem.html [hunk 26]
```diff
@@ -0,0 +1,4 @@
+<a href='<%= "http://purl.bioontology.org/ontology/OMIM/" + data.omim_id %>' "target"="_blank"><%= data.name %></a>
```
Comment (naegelyd) [inline varify/static/templates/modals/diagnosesItem.html]: See notes above about formatting this link

File: varify/static/templates/modals/diagnoses.html [hunk 27]
```diff
@@ -0,0 +1,5 @@
+<h6><%= data.name %></h6>
+<ul data-target=items></ul>
+<% if (!data.collection) { %>
```
Comment (naegelyd) [inline varify/static/templates/modals/diagnoses.html]: See note above about emptyView

File: varify/static/templates/modals/diagnosesItem.html [hunk 28]
```diff
@@ -0,0 +1,4 @@
...
+<% if (data.priority) { %>
+    <span class="badge badge-important"> <%= data.priority %> </span>
```
Comment (naegelyd) [inline varify/static/templates/modals/diagnosesItem.html]: Single quores instead of double quotes

File: varify/static/templates/modals/phenotypes.html [hunk 29]
```diff
@@ -4,25 +4,65 @@
     <h3 data-target=header-label></h3>
 </div>
 
-<div class=modal-body>
+<div class='modal-body phenotype-detail-modal'>
     <div class=span12>
         <div data-target=warning data-action=hide-on-retrieve class='alert alert-warning alert-block hide'></div>
         <div data-target=error data-action=hide-on-retrieve class='alert alert-error alert-block hide'></div>
         <div data-target=loading data-action=hide-on-retrieve>
             <h5><i class="icon-spinner icon-spin"></i> Loading Phenotypes...</h5>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html]: single quotes

File: varify/static/templates/modals/phenotypes.html [hunk 30]
```diff
@@ -4,25 +4,65 @@
...
+            <div class='row-fluid horizontal-list-display' data-target=update-times>
+                <div class=span6><h6>Phenotypes Updated:</h6><%= data.last_modified %></div>
+                <div class=span6><h6>Phenotypes Modified:</h6><%= data.phenotype_modified %></div>
+            </div>
+            <div class='row-fluid pedigree-display'>
+                <h6>Pedigree:</h6>
+                <a href='<%= data.pedigree %>' class=pull-left data-target=pedigree data-action=hide-on-retrieve>View Full Size Pedigree</a>
+                <br>
+                <a href='<%= data.pedigree %>' data-target=pedigree data-action=hide-on-retrieve>
+                    <img data-target=thumbnail src='<%= data.pedigree_thumbnail %>' alt='None' height="180" width="150">
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html]: single quotes

Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html]: Add more useful alt text

File: varify/static/templates/modals/phenotypes.html [hunk 31]
```diff
@@ -4,25 +4,65 @@
...
+
+            <div class='row-fluid horizontal-list-display' data-target=update-times>
+                <div class=span6><h6>Phenotypes Updated:</h6><%= data.last_modified %></div>
+                <div class=span6><h6>Phenotypes Modified:</h6><%= data.phenotype_modified %></div>
+            </div>
+            <div class='row-fluid pedigree-display'>
+                <h6>Pedigree:</h6>
+                <a href='<%= data.pedigree %>' class=pull-left data-target=pedigree data-action=hide-on-retrieve>View Full Size Pedigree</a>
+                <br>
+                <a href='<%= data.pedigree %>' data-target=pedigree data-action=hide-on-retrieve>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html]: Line 56 also specifies `data-target=pedigree`. Is that OK in this case?

File: varify/static/js/src/ui/modals/details/Annotations.js (hunk 15 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/details/Annotations.js]: it adds a new option called collection which (false if there is no collection) displays the empty collection message on the templates

File: varify/static/js/src/ui/modals/phenotype.js (hunk 21 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js]: oops. the method name should be changed. This actually returns a collection.

File: varify/static/js/src/ui/modals/details/Annotations.js (hunk 15 above)
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details/Annotations.js]: Use emptyView instead. The link is provided in a comment below

File: varify/static/js/src/ui/modals/details.js (hunk 9 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/details.js]: phenotype.js is already present in this folder

File: varify/static/js/src/ui/modals/phenotype.js (hunk 19 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js]: it returns the attributes instead of the whole model? Or else how would data.[key] work in the template? Or is this done automatically?

File: varify/static/js/src/ui/modals/details.js (hunk 9 above)
Comment (naegelyd) [inline varify/static/js/src/ui/modals/details.js]: The files in the details folder(which should be renamed to phenotype) are not needed anywhere outside of phenotype.js. Remove details.js and just import the files you need directly in phenotype.js

File: varify/static/js/src/ui/modals/phenotype.js (hunk 19 above)
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: The result of serializeData is passed directly to the template that is rendered. This is noted in the [marionette documentation](https://github.com/marionettejs/backbone.marionette/blob/master/docs/marionette.itemview.md#itemview-serializedata).

File: varify/static/js/src/main.js [hunk 32]
```diff
@@ -23,14 +23,20 @@ require({
     'project/csrf',
     'tpl!../../templates/tables/header.html',
     'tpl!../../templates/modals/result.html',
-    'tpl!../../templates/modals/phenotypes.html',
+    'tpl!../../templates/modals/phenotype/phenotypes.html',
```
Comment (naegelyd) [inline varify/static/js/src/main.js]: Keep phenotypes.html at the /modals/ level

File: varify/static/js/src/main.js [hunk 33]
```diff
@@ -57,7 +63,12 @@ require({
     c.templates.set('varify/export/dialog', exportDialog);
     c.templates.set('varify/tables/header', header);
     c.templates.set('varify/modals/result', result);
-    c.templates.set('varify/modals/phenotype', phenotype);
+    c.templates.set('varify/modals/phenotype/phenotype', phenotype);
```
Comment (naegelyd) [inline varify/static/js/src/main.js]: keep phenotype at the /modals/ level

File: varify/static/js/src/ui/modals/phenotype.js [hunk 34]
```diff
@@ -60,46 +69,24 @@ define([
...
-                html.push('</ul>');
             }
-            else {
-                html.push('<span class=muted>No HPO annotations</span>');
-            }
-
-            html.push('</div>');
-
-            return html.join('');
+            return new Backbone.Collection(collection);
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: What's the purpose for all this? Couldn't you just pass `new Backbone.Collection(annotations)` to the annotations CompositeView?

File: varify/static/js/src/ui/modals/phenotype.js [hunk 35]
```diff
@@ -118,106 +105,80 @@ define([
...
                 }
-
-                html.push('</ul>');
-
-                return html.join('');
-            }
-            else {
-                return '<span class=muted>No diagnoses</span>';
             }
+            return new Backbone.Collection(collection);
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: See comment above about collection construction

File: varify/static/js/src/ui/modals/phenotype.js [hunk 36]
```diff
@@ -118,106 +105,80 @@ define([
...
+        onFetchError: function() {
+            delete this.request;
 
-            html.push('<div class=span4>');
-            html.push('<h6>Confirmed Diagnoses:</h6>');
-            html.push(this._renderDiagnoses(attr.confirmedDiagnoses));
-            html.push('</div>');
+            this.ui.loading.hide();
+            this.ui.error.html('There was an error retrieving the phenotypes.');
+            this.ui.error.show();
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Chain this with the previous error method so it would be

``` javascript
this.ui.error.html('...').show();
```

File: varify/static/js/src/ui/modals/phenotype.js [hunk 37]
```diff
@@ -118,106 +105,80 @@ define([
...
-        renderUpdateTimes: function(attr) {
-            var html = [];
+            var attr = {};
 
-            html.push('<div class=span6>');
-            html.push('<h6>Phenotypes Updated: </h6>' + attr.last_modified);    // jshint ignore: line
-            html.push('</div>');
+            // Initially, the model is not set to anything. It is only after the
+            // request for the phenotype is successful that we have a model.
+            if (this.model) {attr=this.model.attributes;} else return;
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Remove unnecessary case:

``` javascript
if (!this.model) return;

// Remove attr altogether and just use this.model.get('<property_name>') to reference the data you want
```

File: varify/static/js/src/ui/modals/phenotype.js [hunk 38]
```diff
@@ -118,106 +105,80 @@ define([
...
-            html.push('<div class=span6>');
-            html.push('<h6>Rankings Updated: </h6>' + attr.phenotype_modified); // jshint ignore: line
-            html.push('</div>');
 
-            return html.join('');
-        },
+            // Render each region
+            this.annotations.show(new annotations.Annotations({
+                name: "HPO Annotations",
+                collection: this.populateHPOcollection(attr.hpoAnnotations),
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: See comments above, why can't this just be:

`collection: new Backbone.Collection(this.model.get('hpoAnnotations')),`

File: varify/static/js/src/ui/modals/phenotype.js [hunk 39]
```diff
@@ -118,106 +105,80 @@ define([
...
+                itemViewOptions: {'name': 'HPO Annotations'}
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Remove this here. In `itemViewOptions` in the compositeview, just reference the `name` option. You are essentially setting name twice here and in all the 3 diagnoses below.

File: varify/static/js/src/ui/modals/phenotype.js [hunk 40]
```diff
@@ -264,7 +229,7 @@ define([
 
                 this.ui.error.show();
             }
-        },
+         },
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Revert this change

File: varify/static/templates/modals/phenotype/empty.html [hunk 41]
```diff
@@ -0,0 +1 @@
+<%= "No " + data.name %>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotype/empty.html]: move the text out of the template tag:

`No <%= data.name %>`

File: varify/static/templates/modals/phenotype/diagnosesItem.html [hunk 42]
```diff
@@ -0,0 +1,4 @@
+<a href='<%= "http://purl.bioontology.org/ontology/OMIM/" + data.omim_id %>' target=_blank><%= data.name %></a>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotype/diagnosesItem.html]: Move the text out of the template tag. See [example here](https://github.com/cbmi/varify/blob/279478179116cab5ab27aaaf73d6d037b8150ded/varify/static/templates/variant/article-item.html#L1).

THis change will need to be applied to all these templates.

File: varify/static/js/src/ui/modals/phenotype/diagnoses.js [hunk 43]
```diff
@@ -0,0 +1,45 @@
...
+        itemView: DiagnoseItem,
+
+        emptyView: DiagnosesEmptyView,
+
+        itemViewContainer: '[data-target=items]',
+
+        serializeData: function() {
+            return {'name': this.options.name};
+        },
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/diagnoses.js:39]: Remove blank line

File: varify/static/js/src/ui/modals/phenotype/diagnoses.js [hunk 44]
```diff
@@ -0,0 +1,45 @@
...
+        itemView: DiagnoseItem,
+
+        emptyView: DiagnosesEmptyView,
+
+        itemViewContainer: '[data-target=items]',
+
+        serializeData: function() {
+            return {'name': this.options.name};
+        },
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/diagnoses.js:38]: remove trailing comma

File: varify/static/js/src/ui/modals/phenotype.js (hunk 39 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js]: this.options.name is undefined when accessed from the itemViewOptions in the composite view

Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: How are you setting the itemViewOptions in the composite view? Are you returning the options from a function or just setting them directly? 

``` javascript
itemViewOptions: function(model, index) {
     return {
         ...
     };
}
```

or

``` javascript
itemViewOptions: {
    ...
}
```

Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js]: I was setting them directly. When I instantiate the composite view I don't pass in a model. So I think the itemViewOptions it take 
`itemViewOptions: function(name) { 
        ....
}`
?
Though the documentation says that itemViewOptions takes only a `model` or `item` as parameters

Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: The model here is the model for the itemview, the same goes for the index. See the [marionette docs on collectionview itemviewoptions](https://github.com/marionettejs/backbone.marionette/blob/master/docs/marionette.collectionview.md#collectionviews-itemviewoptions) for more details.

File: varify/static/js/src/ui/modals/phenotype.js [hunk 45]
```diff
@@ -260,9 +192,7 @@ define([
                 this.ui.headerLabel.text('Phenotypes');
 
                 if (samples.length === 0) this.ui.error.html(this.alerts.missingSample);
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: Make this a standard if/else block of the form

``` javascript
if (sample.length === 0) {
...
}
else {
...
}
```

File: varify/static/js/src/ui/modals/phenotype.js [hunk 46]
```diff
@@ -2,32 +2,41 @@
...
+    './phenotype/diagnoses',
+    './phenotype/annotations',
     '../../models',
     '../../utils'
-], function(_, Marionette, models, utils) {
+], function(_, Backbone, Marionette, diagnoses, annotations, models, utils) {
 
-    var Phenotype = Marionette.ItemView.extend({
-        className: 'modal hide',
+       var Phenotype = Marionette.Layout.extend({
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: fix indent on this line

File: varify/static/js/src/ui/modals/phenotype.js [hunk 47]
```diff
@@ -118,106 +92,60 @@ define([
...
+            delete this.request;
 
-                return html.join('');
-            }
-            else {
-                return '<span class=muted>No diagnoses</span>';
-            }
+            this.ui.loading.hide();
+            this.ui.error.html('There was an error retrieving the phenotypes.');
+            this.ui.error.show();
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: [Chain](http://www.w3schools.com/jquery/jquery_chaining.asp) this with the .html() call above

File: varify/static/js/src/ui/modals/phenotype.js [hunk 48]
```diff
@@ -118,106 +92,60 @@ define([
...
-            html.push(this._renderDiagnoses(attr.suspectedDiagnoses));
-            html.push('</div>');
-
-            html.push('<div class=span4>');
-            html.push('<h6>Ruled Out Diagnoses:</h6>');
-            html.push(this._renderDiagnoses(attr.ruledOutDiagnoses));
-            html.push('</div>');
-
-            return html.join('');
+        recalculate: function() {
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js:102]: Since recalculate does nothing except execute a single line, replace the `recalculate()` calls with `this.retrievePhenotypes(true)`

File: varify/static/js/src/ui/modals/phenotype.js [hunk 49]
```diff
@@ -279,11 +213,9 @@ define([
 
             this.$el.modal('hide');
         }
-
     });
 
     return {
         Phenotype: Phenotype
     };
-
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js:288]: Restore this blank line

File: varify/static/js/src/ui/modals/phenotype/annotations.js [hunk 50]
```diff
@@ -0,0 +1,44 @@
...
+        serializeData: function() {
+            return {'name': this.options.name};
+        }
+    });
+
+    var AnnotationItem = Marionette.ItemView.extend({
+        tagName: 'li',
+
+        template: 'varify/modals/phenotype/annotationsItem',
+
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/annotations.js]: Remove blank line

File: varify/static/js/src/ui/modals/phenotype/annotations.js [hunk 51]
```diff
@@ -0,0 +1,44 @@
...
+        serializeData: function() {
+            return {'name': this.options.name};
+        }
+    });
+
+    var AnnotationItem = Marionette.ItemView.extend({
+        tagName: 'li',
+
+        template: 'varify/modals/phenotype/annotationsItem',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/annotations.js]: remove trailing comma

File: varify/static/js/src/ui/modals/phenotype/diagnoses.js [hunk 52]
```diff
@@ -0,0 +1,48 @@
...
+    var DiagnosesEmptyView = Marionette.ItemView.extend({
+        tagName: 'span',
+
+        className: 'muted',
+
+        template: 'varify/modals/phenotype/empty',
+
+        serializeData: function() {
+            return {'name': this.options.name};
+        },
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/diagnoses.js]: remove trailing comma

File: varify/static/js/src/ui/modals/phenotype/diagnoses.js [hunk 53]
```diff
@@ -0,0 +1,48 @@
...
+
+        itemViewOptions: function() {
+            return {'name': this.options.name};
+        }
+    });
+
+    return {
+        Diagnoses: Diagnoses
+    };
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype/diagnoses.js:47]: add blank line after

File: varify/static/js/src/ui/modals/phenotype.js [hunk 54]
```diff
@@ -2,32 +2,41 @@
...
+    './phenotype/annotations',
     '../../models',
     '../../utils'
-], function(_, Marionette, models, utils) {
+], function(_, Backbone, Marionette, diagnoses, annotations, models, utils) {
 
-    var Phenotype = Marionette.ItemView.extend({
-        className: 'modal hide',
+       var Phenotype = Marionette.Layout.extend({
+        className: 'phenotype-detail-modal modal hide',
```
Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js]: This is [specified in the HTML](https://github.com/cbmi/varify/pull/302/files#diff-646cb9f086e85df2a623abac1f5b618bR7). Pick one place and only define it there. No need to duplicate this.

File: varify/static/templates/modals/phenotypes.html [hunk 55]
```diff
@@ -4,25 +4,65 @@
...
+                <div class=span4 data-target=suspected-diagnoses></div>
+                <div class=span4 data-target=ruled-out-diagnoses></div>
+            </div>
+
+            <div class='row-fluid horizontal-list-display' data-target=findings>
+                <div class=span9><h6>Imaging Studies:</h6><%= data.imaging_studies %></div>
+                <div class=span9><h6>Procedures:</h6><%= data.procedures %></div>
+                <div class=span9><h6>Significant PE findings:</h6><%= data.significant_pe_findings %></div>
+            </div>
+            <div class='row-fluid horizontal-list-display' data-target=history>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html:44]: blank line above

File: varify/static/templates/modals/phenotypes.html [hunk 56]
```diff
@@ -4,25 +4,65 @@
...
+                <div class=span7><h6>Developmental hx:</h6><%= data.developmental_hx %></div>
+                <div class=span7><h6>Family hx:</h6><%= data.family_hx %></div>
+                <div class=span8><h6>Medical hx:</h6><%= data.medical_hx %></div>
+            </div>
+
+            <div class='row-fluid horizontal-list-display' data-target=update-times>
+                <div class=span6><h6>Phenotypes Updated:</h6><%= data.last_modified %></div>
+                <div class=span6><h6>Phenotypes Modified:</h6><%= data.phenotype_modified %></div>
+            </div>
+            <div class='row-fluid pedigree-display'>
```
Comment (naegelyd) [inline varify/static/templates/modals/phenotypes.html:56]: blank line above

File: varify/static/js/src/ui/modals/phenotype.js (hunk 48 above)
Comment (hassanNS) [inline varify/static/js/src/ui/modals/phenotype.js:102]: recalculate is a function bound to the recalculate button click event. It's not called anywhere else and there are problems when trying to bind retrivePhenotype(true) to the click event.

Comment (naegelyd) [inline varify/static/js/src/ui/modals/phenotype.js:102]: Eh, just leave this as is for now
